### PR TITLE
[main] Add total memory to device metadata

### PR DESCRIPTION
**Pull request by onalante-msft**
Total memory is just as important as CPU core count in terms of providing a picture of host system properties. For this reason, we now provide the `total_memory` metric through the `system_info` API directly, where it was previously only available through the `system_resources` API.

Updated tests:
- `Microsoft.Azure.Devices.Edge.Agent.Core.Test/SystemInfoTests.cs`: include `total_memory`/`TotalMemory` in serializations.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [x] local testing done.

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs [hunk 1]
```diff
@@ -163,15 +163,15 @@ public static async Task<int> MainAsync(IConfiguration configuration)
                     case Constants.IotedgedMode:
                         string managementUri = configuration.GetValue<string>(Constants.EdgeletManagementUriVariableName);
                         string workloadUri = configuration.GetValue<string>(Constants.EdgeletWorkloadUriVariableName);
-                        bool disableDeviceAnalyticsTelemetry = configuration.GetValue<bool>("DisableDeviceAnalyticsTelemetry", false);
+                        bool disableDeviceAnalyticsMetadata = configuration.GetValue<bool?>("DisableDeviceAnalyticsMetadata") ?? configuration.GetValue<bool>("DisableDeviceAnalyticsTelemetry", false);
```
Comment (varunpuranik) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs:166]: Did you test if this logic works as expected?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/SystemInfo.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Core
 
     public class SystemInfo
     {
-        public SystemInfo(string operatingSystemType, string architecture, string version, ProvisioningInfo provisioning, string serverVersion, string kernelVersion, string operatingSystem, int numCpus, string virtualized, IReadOnlyDictionary<string, object> additionalProperties)
+        public SystemInfo(string operatingSystemType, string architecture, string version, ProvisioningInfo provisioning, string serverVersion, string kernelVersion, string operatingSystem, int numCpus, int totalMemory, string virtualized, IReadOnlyDictionary<string, object> additionalProperties)
```
Comment (micahl) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/SystemInfo.cs:14]: MaxValue for an `int` (which IIRC aliases to Int32) is 2,147,483,647.  So... would we potentially have overflow on machines with more memory?

Comment (onalante-msft) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/SystemInfo.cs]: Great catch, thank you. Fixed in 41aa35174cc476af37b8a174a1b051ffbc764a48.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs (hunk 1 above)
Comment (onalante-msft) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs:166]: Yes, I have a local project that I used to confirm this behavior.

File: edgelet/edgelet-docker/src/runtime.rs [hunk 3]
```diff
@@ -525,6 +525,12 @@ where
     async fn system_info(&self) -> anyhow::Result<CoreSystemInfo> {
         log::info!("Querying system info...");
 
+        let total_memory = {
+            let mut system_resources = self.system_resources.as_ref().lock().await;
+            system_resources.refresh_memory();
+            system_resources.total_memory() * 1000
```
Comment (arsing) [inline edgelet/edgelet-docker/src/runtime.rs:531]: The implementation uses the "MemTotal" line from `/proc/meminfo`, which is in KiB (even though the file confusingly calls it "kB"), so this should be `* 1024`

**Review by arsing — CHANGES_REQUESTED** [review]

File: edgelet/edgelet-docker/src/runtime.rs (hunk 3 above)
Comment (onalante-msft) [reply edgelet/edgelet-docker/src/runtime.rs:531]: Very confusing, will fix.

Comment (onalante-msft) [reply edgelet/edgelet-docker/src/runtime.rs]: cb209a831c282bc4c42f72b15c459f931b9638f7

Comment (onalante-msft) [reply edgelet/edgelet-docker/src/runtime.rs]: Fix for two other memory query sites: af40269b2fcf231f3c0553f1d68e7db7cb8b891d

**Review by arsing — APPROVED** [review]

**Review by ancaantochi — DISMISSED** [review]
Please test with updated edgeAgent and edgeHub and older edged and then updated edged and older edgeHub and edgeAgent images to make sure versions compatibility is still good.

**Review by arsing — APPROVED** [review]

**Pull request merged**